app/assets/javascripts/comfortable_mexican_sofa/admin/modules/CategoryExpand.js

```diff
@@ -0,0 +1,76 @@
+define(['jquery','DoughBaseComponent'], function ($, DoughBaseComponent) {
+'use strict';
+
+  var defaultConfig = {
+    categoriesCharacterCount: 20
+  };
+  var CategoryExpandProto;
+
+  function CategoryExpand($el, config) {
+    CategoryExpand.baseConstructor.call(this, $el, config, defaultConfig);
+  }
+
+  DoughBaseComponent.extend(CategoryExpand);
+
+  CategoryExpand.componentName = 'CategoryExpand';
+
+  CategoryExpandProto = CategoryExpand.prototype;
+
+  CategoryExpandProto.init = function(initialised) {
+
+    this.addExpanders();
+    this.addListeners();
+
+    this._initialisedSuccess(initialised);
+    return this;
+
+  };
+
+  CategoryExpandProto.addListeners = function() {
+
+    $('#filter_form').on('ajax:complete', $.proxy(function() {
+      this.addExpanders();
+    }, this));
+
+  };
+
+  CategoryExpandProto.addExpanders = function() {
+
+    var self = this;
+
+    this.$el.find('.categories').each(function() {
+
+      var categories = $(this).text(),
+          categoryCount = categories.split(',').length - 1,
+          categoriesCharacterCount = self.config.categoriesCharacterCount;
+
+      if (categories.indexOf(',') >= 1) {
+        $(this).html(
+          categories.slice(0,categoriesCharacterCount)+
+          '<a href="#" class="more"> +' + categoryCount + ' see all</a>'+
+          '<span style="display:none;" class="allCategories">'+ categories.slice(categoriesCharacterCount ,categories.length)+' <a href="#" class="less">Hide</a></span>'
+        );
+      };
+
+    });
+
+    this.$el.find('.more').click(function(event) {
+      event.preventDefault();
+      $(this).hide();
+      $(this).parent().find('.allCategories').show();
+    });
+
+    this.$el.find('.less').click(function(event) {
+      event.preventDefault();
+      $(this).parent().hide();
+      $(this).closest('.categories').find('.more').show();
+    });
+
```

Comment:
It would be nice to have these bound to $more and $less variables and/or use delegation to handle these click events. It would also be good to remove the .parent() references however in some situations it can be useful. Perhaps we can create a $categories variable as well.